devices/nfc/nfc.js

```diff
@@ -51,6 +51,11 @@ nfc.on("reader", (reader) => {
 
     const barcode = await reader.transmit(packet, 40, 2);
 
+    // Check command processing status indicates success
+    if (!barcode.subarray[-2].equals(Buffer.from('9000', 'hex'))) {
+      throw new Error("NFC card did not scan correctly");
```

Comment:
Sadly, throwing errors only does auto-magical things when you're inside a request handler: `socket.handle('foo', () => {... throw ..})`. Long story short, if I make a request to an external service, and the handler throws, then `reconnecting-socket` will send that error back to the client. In the POS, we have an error handler that catches these errors and displays them: https://github.com/chezbob/chezbob/blob/93eb1f4544536e0a38f644d9bc0df248ae054e34/web/static/internal/apps/pos/pos.js#L38.  If we want to display an error message on the POS, without the POS having issued a request, we'll need a new message type:

```js
socket.send({
  header: {
    to: DESTINATION_IDENT,
    type: "nfc_scan_error",
  },
  body: {}
});
```

And then we'd need  a handler on the POS side. I'd just install a global handler in `pos.js`:

```js
socket.on('nfc_scan_error', () => {
  window.mode.set_error('Error reading NFC card. Try again");
  // Even though it's an error, it's an indication the user is actively using the system so bump the timeout
  window.mode.bumpTimeout && window.mode.bumpTimeout();
});
```
 
Alternately, we could create a generic "display_error" message to send the POS but that would couple the `nfc` service to the `pos` which is unnecessary.